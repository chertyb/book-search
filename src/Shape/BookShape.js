import PropTypes from 'prop-types';

const BookShape = PropTypes.shape({
  id: PropTypes.string,
  title: PropTypes.string,
  authors: PropTypes.string,
  categories: PropTypes.string,
  description: PropTypes.string,
  thumbnail: PropTypes.string,
});

export const defaultBookShape = {
  id: '',
  title: '',
  authors: '',
  categories: '',
  description: '',
  thumbnail: '',
};

export default BookShape;
