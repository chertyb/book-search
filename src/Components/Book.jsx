import React from 'react';
import BookShape, { defaultBookShape } from '../Shape/BookShape';

const Book = (props) => {
  const { data } = props;

  return (
    <div className="book">
      <img src={data.thumbnail} alt="thumbnail" />
      <p title="Title">
        {'Title: '}
        {data.title}
      </p>
      <p title="Authors">
        {'Authors: '}
        {data.authors}
      </p>
      <p title="Categories">
        {'Categories: '}
        {data.categories}
      </p>
      <p title="Description">
        {'Description: '}
        {data.description}
      </p>
    </div>
  );
};

Book.propTypes = {
  data: BookShape,
};

Book.defaultProps = {
  data: defaultBookShape,
};

export default Book;
