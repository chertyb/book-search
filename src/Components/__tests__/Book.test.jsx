import React from 'react';
import ReactDom from 'react-dom';
import { shallow } from '../../enzyme';
import Book from '../Book';
import { defaultBookShape } from '../../Shape/BookShape';

describe('Book component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(<Book />, div);
    expect(div).toMatchSnapshot();
    ReactDom.unmountComponentAtNode(div);
  });

  it('should display default props value', () => {
    const wrapper = shallow(<Book />);
    expect(wrapper.find('p[title="Title"]').text()).toEqual(`Title: ${defaultBookShape.id}`);
    expect(wrapper.find('p[title="Authors"]').text()).toEqual(`Authors: ${defaultBookShape.authors}`);
    expect(wrapper.find('p[title="Categories"]').text()).toEqual(`Categories: ${defaultBookShape.categories}`);
    expect(wrapper.find('p[title="Description"]').text()).toEqual(`Description: ${defaultBookShape.description}`);
    expect(wrapper.find('img').prop('src')).toEqual(defaultBookShape.id);
  });

  it('should display custom props value', () => {
    const book = {
      id: 'id',
      title: 'title',
      authors: 'authors',
      categories: 'categories',
      description: 'description',
      thumbnail: 'thumbnail',
    };
    const wrapper = shallow(<Book data={book} />);
    expect(wrapper.find('p[title="Title"]').text()).toEqual(`Title: ${book.title}`);
    expect(wrapper.find('p[title="Authors"]').text()).toEqual(`Authors: ${book.authors}`);
    expect(wrapper.find('p[title="Categories"]').text()).toEqual(`Categories: ${book.categories}`);
    expect(wrapper.find('p[title="Description"]').text()).toEqual(`Description: ${book.description}`);
    expect(wrapper.find('img').prop('src')).toEqual(book.thumbnail);
  });
});
