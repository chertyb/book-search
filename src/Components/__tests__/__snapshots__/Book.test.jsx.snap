// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book component renders without crashing 1`] = `
<div>
  <div
    class="book"
  >
    <img
      alt="thumbnail"
      src=""
    />
    <p
      title="Title"
    >
      Title: 
      
    </p>
    <p
      title="Authors"
    >
      Authors: 
      
    </p>
    <p
      title="Categories"
    >
      Categories: 
      
    </p>
    <p
      title="Description"
    >
      Description: 
      
    </p>
  </div>
</div>
`;
